packages/client/components/ReflectionGroup/useSpotlightVisibleReflections.ts

```diff
@@ -0,0 +1,28 @@
+import {useMemo} from 'react'
+import {ReflectionGroup_reflectionGroup} from '~/__generated__/ReflectionGroup_reflectionGroup.graphql'
+
+const useSpotlightVisibleReflections = (
+  reflections: ReflectionGroup_reflectionGroup['reflections'],
+  spotlightSearchQuery: string | null,
+  reflectionIdsToHide?: string[] | null
+) => {
+  return useMemo(() => {
+    const visibleReflections = reflections.filter(({id}) => !reflectionIdsToHide?.includes(id))
+
+    if (spotlightSearchQuery != null && visibleReflections.length > 1) {
+      const spotlightSearchQueryLower = spotlightSearchQuery.toLowerCase()
+      const matchIndex = visibleReflections.findIndex((reflection) => {
+        const textLower = reflection.plaintextContent.toLowerCase()
+        return textLower.indexOf(spotlightSearchQueryLower) !== -1
+      })
+
+      if (matchIndex > 0) {
+        visibleReflections.unshift(visibleReflections.splice(matchIndex, 1)[0])
```

Comment:
NIT: I'd find it easier to read if the splice was moved out of the unshift function. Something like: 
```
const matchingReflection = visibleReflections.splice(matchIndex, 1)[0]
visibleReflections.unshift(matchingReflection)
```